froide/organization/views.py

```diff
@@ -0,0 +1,208 @@
+            organization__organizationmembership__status=OrganizationMembership.STATUS.ACTIVE,
+        ).exclude(user=self.request.user)
+
+
+class DeleteOrganizationMemberView(
+    LoginRequiredMixin, ManageOrganizationMemberMixin, DetailView
+):
+    def get(self, request, *args, **kwargs):
+        return redirect(
+            reverse("organization-update", slug=self.get_object().organization.slug)
+        )
+
+    def post(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        organization = self.object.organization
+        self.object.delete()
+        return redirect(
+            reverse("organization-update", kwargs={"slug": organization.slug})
+        )
+
+
+class ChangeOrganizationMemberRoleView(
+    LoginRequiredMixin, ManageOrganizationMemberMixin, UpdateView
+):
+    form_class = OrganizationMemberChangeRoleForm
+
+    def get(self, request, *args, **kwargs):
+        return redirect(
+            reverse(
+                "organization-update",
+                kwargs={"slug": self.get_object().organization.slug},
+            )
+        )
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        member = kwargs["instance"]
+        owner = member.organization.organizationmembership_set.get(
+            user=self.request.user
+        )
+        kwargs["owner"] = owner
+        return kwargs
+
+    def get_success_url(self) -> str:
+        return reverse(
+            "organization-update", kwargs={"slug": self.get_object().organization.slug}
+        )
+
+
+class InviteOrganizationMemberView(LoginRequiredMixin, UpdateView):
+    form_class = OrganizationInviteForm
+    template_name = "organization/organization_detail.html"
+
+    def get_queryset(self):
+        return Organization.objects.get_owner_organizations(self.request.user).filter(
+            slug=self.kwargs["slug"]
+        )
+
+    def get(self, request, *args, **kwargs):
+        return redirect(self.get_object())
+
+    def form_valid(self, form):
+        """
+        If the form is valid, save the associated model.
+        """
+        self.object = form.save(self.request.user)
+        return redirect(
+            reverse("organization-update", kwargs={"slug": self.object.slug})
+        )
+
+    def form_invalid(self, form):
+        return redirect(
+            reverse("organization-update", kwargs={"slug": self.object.slug})
+        )
+
+
+class JoinOrganizationView(LoginRequiredMixin, DetailView):
+    template_name = "organization/organization_join.html"
+
+    def get_queryset(self):
+        return OrganizationMembership.objects.filter(
+            organization__slug=self.kwargs["slug"],
+            status=OrganizationMembership.STATUS.INVITED,
+        )
+
+    def get_object(self):
+        member = super().get_object()
+        service = OrganizationService(member)
+        if not service.check_invite_secret(self.kwargs["secret"]):
+            raise Http404
```

Comment:
Maybe something friendlier? Like with `already_member`? Could be that a link was borked in an email.

Reply:
This would mean that we show a different response if the invite-id is invalid (404) vs if the secret is wrong (friendlier error message). Do we want to do that?